LiveSplit.ActionHenk.asl

```diff
@@ -1,286 +1,270 @@
-                               ((old.gotta_ghost_fast == 4 && old.wicked_waves     == 4 && old.cursed_curves     == 4 && old.deadly_drops  == 4 && old.tricks_treats    == 4 && old.kentony == 1         && old.zombie_neil == 1)     ? 1 : 0) +
-                               ((old.rise_n_shine     == 4 && old.pipe_n_slide     == 4 && old.quick_tricks      == 4 && old.wave_rider    == 4 && old.down_the_tube    == 4 && old.lifegaurd_betsy == 1 && old.summer_henk == 1)     ? 1 : 0) +
-                               ((old.deep_dive        == 4 && old.the_plunger      == 4 && old.the_big_climb     == 4 && old.sad_snails    == 4 && old.leap_of_faith    == 4 && old.cedar == 1           && old.rastafro == 1)        ? 1 : 0) +
-                               ((old.full_swing_ahead == 4 && old.hook_maze        == 4 && old.throwback         == 4 && old.flappy_swing  == 4 && old.right_round_baby == 4 && old.jungle_cedar == 1    && old.tribal_cedar == 1)    ? 1 : 0) +
-                               ((old.sick_burn        == 4 && old.full_stop        == 4 && old.the_wall          == 4 && old.spinebreaker  == 4 && old.pinball          == 4 && old.kentinator == 1      && old.afronaut == 1)        ? 1 : 0) +
-                               ((old.transition_kings == 4 && old.hardcore_hooks   == 4 && old.hi_speed_hysteria == 4 && old.ultimate_test == 4 && old.way_of_the_ninja == 4 && old.nineties_henk == 1   && old.action_hank == 1)     ? 1 : 0) +
-                               ((old.the_highway      == 4 && old.back_and_forth   == 4 && old.slidey_slidey     == 4 && old.bumper_jumper == 4 && old.the_zigzag       == 4 && old.santa_henk == 1      && old.henkdolph == 1)       ? 1 : 0);
-
-  return true;
+    vars.watchers.UpdateAll(game);
+
+    // Update old local variables
+    vars.oldSumMedals = vars.curSumMedals;
+    vars.oldFullBatches = vars.curFullBatches;
+    vars.oldRainbowBatches = vars.curRainbowBatches;
+    vars.oldCompletedBatches = vars.curCompletedBatches;
+    
+    // Update when the medal count change in the current level
+  if(vars.levelCode.Old != 0 && vars.bestMedal.Old < vars.bestMedal.Current) {
+        int indexOfLevel = Array.IndexOf(vars.levelsCode[vars.lookingAtBatchNum.Current], vars.levelCode.Current);
+        vars.levelsMedals[vars.lookingAtBatchNum.Current][indexOfLevel] = vars.bestMedal.Current;
+        vars.curSumMedals += vars.bestMedal.Current-vars.bestMedal.Old;
+
+        // Recalculation of batches
+        bool batchIsFull, batchIsRainbow, batchIsCompleted;
+        vars.curFullBatches = vars.curRainbowBatches = vars.curCompletedBatches = 0;
+        for (int batchId = 0; batchId < vars.levelsMedals.Length; batchId++) {
+            int[] batchMedals = vars.levelsMedals[batchId];
+            batchIsFull = batchIsRainbow = batchIsCompleted = true;
+            for (int levelId = 0; levelId < batchMedals.Length; levelId++) {
```

Comment:
nit: do `batchIndex` and `levelIndex` make more sense here? Since `levelId` technically has a distinct meanning from its order in the game?

Reply:
Hmm I don't really understand what you mean by that but if you just want to rename them go ahead. For me, Id is a shortcut for Index.

Reply:
eh, not worth the effort. i don’t really mind, just that levelId has a different global meaning than the value it holds in this loop

Reply:
Oh ok but actually ... no :p.
levelId is only a local variable here (name also used in IsCurrentLevelCompleted function but also local).
The global variable/watch that holds the level index is called levelCode to match the one in game.
Can be confusing though I agree.